autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/BrowserAutofill.kt

```diff
@@ -139,18 +83,44 @@ interface Callback {
      * When this is called, we should present the generated password to the user for them to choose whether to use it or not.
      */
     suspend fun onGeneratedPasswordAvailableToUse(
-        originalUrl: String,
+        autofillUrlRequest: AutofillUrlRequest,
         username: String?,
         generatedPassword: String,
     )
 
     /**
-     * Called when we've been asked which credentials we have available to autofill, but the answer is none.
+     * Called when the user should be shown prompt to choose an email address to use for email protection autofill
+     */
+    fun showNativeChooseEmailAddressPrompt(autofillUrlRequest: AutofillUrlRequest)
+
+    /**
+     * Called when the user should be shown prompt to sign up for Email Protection
      */
-    fun noCredentialsAvailable(originalUrl: String)
+    fun showNativeInContextEmailProtectionSignupPrompt(autofillUrlRequest: AutofillUrlRequest)
 
     /**
      * Called when credentials have been saved, and we want to show the user some visual confirmation.
      */
     fun onCredentialsSaved(savedCredentials: LoginCredentials)
 }
+
+/**
+ * When there is an autofill request to be handled that requires user-interaction, we need to know where the request came from when later responding
+ */
+@Parcelize
+data class AutofillUrlRequest(
```

Comment:
_>why the dialogFragment should know about his object if it's not using it is the question I have_

This is mostly because there are fragment result handlers that will receive the result coming from the dialog fragments, and those result handlers need access to the `requestId` and sometimes also the `requestOrigin` depending on the specific case. So even if the dialog fragment itself doesn't need to use that object, it is expected to receive it and pass it through so that the result handler can access it.

Reply:
_Do we need this model here? / maybe it's not the model but how we are using it across the codebase that makes me think something is off here, or I'm just missing something._

This is a good question, and it does get passed around to a lot of classes. Before, it was just passing around a string. That probably looked better since it was just a string and not a custom class, but I think it's the same amount of coupling on this data that was happening before (i.e., we already had a lot of classes depending on getting this string passed around). Now, we have we an object instead of just that one string.

I'm not sure if it's better to pass that one object type around everywhere vs breaking it apart sooner and only passing the one or two data parts of it where it is needed (e.g., passing `requestOrigin` only). I _think_ it makes sense to contain it into the `AutofillUrlRequest` object (but not 100% on this), as this gives it some consistency and flexibility. 
- e.g., you pass the object and the receiver has everything it needs now and likely will ever need. vs if you provided some cases with just a string (e.g., requestId), or in another case two strings (one for requestId and requestOrigin) and had to change it later you would to change many places to accept both later.

_>in most of them (if not all) we are only making us of the url._

I don't think that's quite right; I think we are using `requestId` more than anything else, but also `requestOrigin` in many cases